src/system-health/health_checker/hardware_checker.py

```diff
@@ -8,7 +8,14 @@ class HardwareChecker(HealthChecker):
     """
     Check system hardware status. For now, it checks ASIC, PSU and fan status.
     """
-    ASIC_TEMPERATURE_KEY = 'TEMPERATURE_INFO|ASIC'
+    import sonic_platform.platform
+    ASIC_TEMPERATURE_KEY = []
+    thermal_list = sonic_platform.platform.Platform().get_chassis().get_all_thermals()
+    for i in range(0, len(thermal_list)):
+         thermal_name = thermal_list[i].get_name()
```

Comment:
what if vendor does not implement get_name? it might raise exception.

Reply:
I believe that the expectation is to have each vendor to implement the functions defined in base class.
To be on the safer side, would put this in try,except  block.